src/audio/module_adapter/module/waves.c

```diff
@@ -665,6 +640,22 @@ static int waves_codec_init(struct processing_module *mod)
 		setup_cfg->avail = true;
 	}
 
+	ret = MaxxEffect_GetMessageMaxSize(waves_codec->effect, &waves_codec->request_max_bytes,
+					      &waves_codec->response_max_bytes);
+
+	if (ret) {
+		comp_err(dev, "cadence_codec_init() MaxxEffect_GetMessageMaxSize returned %d", ret);
+		return -EINVAL;
```

Comment:
looks like a memory leak: `module_free_memory()` is missing. Please, make a clean up path and use `goto` to avoid such leaks in the future.
While at it please also check error handling around line 615
```
if (ret)
	comp_err(dev, "waves_codec_init() failed %d", ret);
```
looks weird - printing a failure message (potentially for a second time) and not returning immediately